src/webapi/xml_http_request.rs

```diff
@@ -0,0 +1,69 @@
+use webcore::value::{
+    Reference,
+    Value,
+};
+use webcore::try_from::TryInto;
+
+/// XMLHttpRequest object to make http requests
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest)
+#[derive(Debug)]
+pub struct XMLHttpRequest {
+    reference: Reference,
+}
+
+impl XMLHttpRequest {
+    /// Creates an empty XMLHttpRequest
+    pub fn new() -> XMLHttpRequest {
+        let reference: Reference = js! {
+            var xmlHttp = new XMLHttpRequest();
+            return xmlHttp;
+        }.try_into().unwrap();
+        XMLHttpRequest {
+            reference,
+        }
+    }
+    /// Sets the callback to call once a request completes
+    /// First argument is the status code, the second argument is the response text
+    /// Makes use of onreadystatechange:
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/onreadystatechange)
+    pub fn set_callback<F:FnMut(u32, String) + 'static>(&self, callback: F) {
```

Comment:
Instead of ad hoc methods like this I'd prefer to do this through `IEventTarget` machinery we already have. (So instead of using this function the user would use `add_event_listener` from the `IEventTarget` interface.) In which case it would also be nice to have an usage example in the doc comments.